### PR TITLE
Created a Samson inbound webhook action

### PR DESCRIPTION
**Pull request by magnetikonline**
- Created a Samson inbound webhook action as a [composite run step](https://github.blog/changelog/2020-08-07-github-actions-composite-run-steps/).
  - Pure YAML/Bash (no Docker needed).
  - Given `webhook-url` is validated for the correct format via Regexp.
  - No need to pass `${{ github.sha }}` - as Action has direct access to this.
- Dropped existing Docker based `samson-create-release` action.
- Added `.editorconfig`

## Example usage:

```yaml
jobs:
  testing:
    steps:
      - name: Samson inbound test
        uses: zendesk/ga/samson-inbound-webhook@v2
        with:
          webhook-url: https://samson.zende.sk/integrations/generic/TOKEN
```

### CODE REVIEW

File: samson-inbound-webhook/action.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  steps:
+    - run: |
+        # validate webhook URL
+        if [[ ! "${{ inputs.webhook-url }}" =~ ^https://[^/]+/integrations/generic/[a-f0-9]{32}$ ]]; then
+          echo "::error::Invalid Samson generic webhook URL format"
+          exit 1
+        fi
+
+        # extract branch name
+        branchName="${{ github.ref }}"
```
Comment (shoukoo) [inline samson-inbound-webhook/action.yaml:24]: Please make branch name as a parameter (having a default as "master"), I know some engineers pass in custom branch name e.g.`master-no-build` to inform Samson not to trigger a deployment.

Also `github.ref` is not a reliable pointer to get branch name, If an action is triggered by `pull request` then the  value contains no branch and uses this format `tefs/pull/<PR no.>/merge`

Comment (magnetikonline) [reply samson-inbound-webhook/action.yaml]: Changed this in ae63b84

File: samson-inbound-webhook/action.yaml [hunk 2]
```diff
@@ -2,6 +2,10 @@ name: Samson inbound webhook
 author: Peter Mescalchin
 description: Send inbound webhook notifications to a Samson server
 inputs:
+  branch:
+    description: Branch name for commit
+    default: master
+    required: true
```
Comment (shoukoo) [inline samson-inbound-webhook/action.yaml:8]: I don't think we need `required`  since we have a dafule value already.

Comment (magnetikonline) [reply samson-inbound-webhook/action.yaml]: Fixed with af22447

**Comment by magnetikonline** [comment]
Just to leave a note @shoukoo I'll do a fix to use either `github.event.pull_request.head.ref` or `github.ref` - as we shouldn't rely on the main branch name being `master` anymore.

**Review by shoukoo — APPROVED** [review]

**Pull request merged**